pkg/oc/generate/app/cmd/newapp.go

```diff
@@ -320,9 +322,14 @@ func (c *AppConfig) AddArguments(args []string) []string {
 
 		switch {
 		case c.tryToAddEnvironmentArguments(s):
+			c.ArgumentClassificationWinner = fmt.Sprintf(winnerFmt, s, "environment")
 		case c.tryToAddSourceArguments(s):
-		case c.tryToAddComponentArguments(s):
+			c.ArgumentClassificationWinner = fmt.Sprintf(winnerFmt, s, "source")
 		case c.tryToAddTemplateArguments(s):
+			c.ArgumentClassificationWinner = fmt.Sprintf(winnerFmt, s, "template")
+		case c.tryToAddComponentArguments(s):
+			// NOTE, component argument classification currently is the most lenient, so we save it for the end
+			c.ArgumentClassificationWinner = fmt.Sprintf(winnerFmt, s, "component")
```

Comment:
"component" is a confusing term here...as a user when i see "the argument x passedt he component classification tests" i don't really know what that means my argument was treated as.  Is there another term we can use?